routers/agents.py

```
@@ -0,0 +1,185 @@
+            raise HTTPException(status_code = 400, detail = "An error occurred: agent is already logged out")
+        
+        await agents_collections.update_one({"_id": agent_record["_id"]}, {"$set": {"is_active": 0}})
+        await agents_collections.update_one({"_id": agent_record["_id"]}, {"$set": {"modified_date": date_time}})
+
+        if configuration_record['auto_assignment']:
+            redis = await get_redis()
+            session_ids = await view_queue(f'{id}:{name}:{email}')
+            await delete_redis(redis, f'{id}:{name}:{email}')
+            
+            for session_id in session_ids:
+                transfer_queue = transfer_queue + f':{bot_id}:{workspace_id}'
```

Ruff (0.8.2)

Local variable `transfer_queue` referenced before assignment

(F823)